src/main/scala/cognite/spark/v1/FlexibleDataModelBaseRelation.scala

```diff
@@ -57,183 +58,164 @@ abstract class FlexibleDataModelBaseRelation(config: RelationConfig, sqlContext:
         Right(
           FilterDefinition.Equals(
-            createNodeOrEdgeCommonAttributeRef(instanceType, "externalId"),
+            createNodeOrEdgeCommonAttributeRef(instanceType, attribute),
             FilterValueDefinition.String(String.valueOf(value))))
       case EqualTo(attribute, value: GenericRowWithSchema)
-          if attribute.equalsIgnoreCase("startNode") && instanceType == InstanceType.Edge =>
-        createEqualsAttributeFilter("startNode", value, instanceType)
-      case EqualTo(attribute, value: GenericRowWithSchema)
-          if attribute.equalsIgnoreCase("endNode") && instanceType == InstanceType.Edge =>
-        createEqualsAttributeFilter("endNode", value, instanceType)
-      case EqualTo(attribute, value: GenericRowWithSchema) if attribute.equalsIgnoreCase("_type") =>
-        createEqualsAttributeFilter("type", value, instanceType)
-      //TODO add exists on view property first
-      case EqualTo(attribute, value: GenericRowWithSchema)
-          if attribute.equalsIgnoreCase("type") && instanceType == InstanceType.Edge =>
-        createEqualsAttributeFilter("type", value, instanceType)
-      case Or(f1, f2) =>
-        Vector(f1, f2)
-          .traverse(toNodeOrEdgeAttributeFilter(instanceType, _))
-          .map(FilterDefinition.Or.apply)
-      case And(f1, f2) =>
-        Vector(f1, f2)
-          .traverse(toNodeOrEdgeAttributeFilter(instanceType, _))
-          .map(FilterDefinition.And.apply)
-      case Not(f) =>
-        toNodeOrEdgeAttributeFilter(instanceType, f)
-          .map(FilterDefinition.Not.apply)
-      //Node types are optional so exists will be automatically checked
-      //Types are a property of the node/edge and should not use the view but directly check on the node
-      case IsNotNull(attribute)
-          if attribute.equalsIgnoreCase("_type") && instanceType == InstanceType.Node =>
-        Right(FilterDefinition.Exists(Seq("node", "type")))
-      case IsNotNull(attribute)
-          if attribute.equalsIgnoreCase("type") && instanceType == InstanceType.Edge =>
-        Right(FilterDefinition.Exists(Seq("edge", "type")))
-      case IsNull(attribute)
-          if attribute.equalsIgnoreCase("_type") && instanceType == InstanceType.Node =>
-        Right(FilterDefinition.Not(FilterDefinition.Exists(Seq("node", "type"))))
-      case IsNull(attribute)
-          if attribute.equalsIgnoreCase("type") && instanceType == InstanceType.Edge =>
-        Right(FilterDefinition.Not(FilterDefinition.Exists(Seq("edge", "type"))))
+          if nodeOrEdgeReferenceAttributes.contains(attribute) =>
+        createEqualsAttributeFilter(createNodeOrEdgeCommonAttributeRef(instanceType, attribute), value)
+      case In(attribute, values) if nodeOrEdgeStringAttributes.contains(attribute) =>
+        toSeqFilterValueDefinition(attribute, values)
+          .filterOrElse(
+            {
+              case FilterValueDefinition.StringList(_) => true
```

Comment:
we start accepting `In` for any reserved property now, should we drop `StringList` requirement?

Reply:
This only accepts in for string attributes (because of the if nodeOrEdgeStringAttributes.contains(attribute) guard) but I could add support for the other ones as well, think I should?

I probably should look at how this sync request is created because I would assume it's standardized somehow otherwize we would encounter "Unsupported node or edge attribute filters" everywhere... though we absorb the exceptions in some places via transforming to optional

Reply:
I guess that's what's happening! When we call the filters in sync we do a .toOption which basically cancel out the "Unsupported node or edge attribute filter" exception.

Reply:
So probably there is tons of filters that are on attributes of views, but they all result in exceptions, then .toOption removes the exceptions into a None then the ++ adds them together and leaves only the filters on instances attributes

Reply:
though it looks broken then, it'd probably give up on having any supported filters if there's unsupported filter somewhere, instead it should probably do what other resources do via `pushdownToFilters` by collecting things looking at logical tree

Reply:
Hmm, not the case everywhere, I'm pretty sure it would indeed throw an error in some, but the one in sync should work: `(filters.map {
      toNodeOrEdgeAttributeFilter(instanceType, _).toOption
    } ++ Array(hasData)).flatten.toSeq` => doesn't that properly discards failing filters?